pkg/spire/spire.go

```diff
@@ -50,23 +71,32 @@ func (sc *SpireServerApiClient) checkClient(ctx context.Context) error {
 }
 
 func (sc *SpireServerApiClient) dial(ctx context.Context) error {
+	sc.ctx = ctx
 	if sc.workloadConn == nil {
 		// Create X509Source
 		source, err := workloadapi.NewX509Source(ctx, workloadapi.WithClientOptions(workloadapi.WithAddr("unix://"+sc.config.SocketPath)))
 		if err != nil {
-			return fmt.Errorf("Unable to create X509Source for SPIFFE client: %w", err)
+			return fmt.Errorf("unable to create X509Source for SPIFFE client: %w", err)
 		}
 		sc.workloadConn = source
 	}
 
+	if sc.workloadAPI == nil {
+		client, err := workloadapi.New(ctx, workloadapi.WithAddr("unix://"+sc.config.SocketPath))
+		if err != nil {
+			return fmt.Errorf("spire workload API not initalized due to error: %s", err)
```

Comment:
nit: use `%w` for wrapping

Reply:
probably need to add this to `checkClient` and add to `Close()`

Reply:
done.